NV-30: frontend auth guard

oppikar opened the pull request:
- setup of authguard class;

oppikar committed:
NV-30:
- setup AuthGuard;

- implement default "home" page;
- implement lazy-loading;
- prohibit routes from loading if not authenticated;

Luca-Guettinger reviewed (changes requested):
maybe das die startseite nicht erricht werden kann und das man auf eine error page redirected wird?

oppikar commented:
> maybe das die startseite nicht erricht werden kann und das man auf eine error page redirected wird?

- Implemented default "Home"
- Implemented route-guard for every Route but 'login' (later there will also be 'register')
- Implemented lazy-loading of routes
- Implemented re-routing towards login if no auth-token is found
